Content.Client/GameObjects/Components/Nutrition/HungerUI.cs

```diff
@@ -0,0 +1,76 @@
+using System;
+using Content.Client.UserInterface;
+using Content.Client.Utility;
+using Content.Shared.GameObjects.Components.Nutrition;
+using Robust.Client.Interfaces.ResourceManagement;
+using Robust.Client.UserInterface.Controls;
+using Robust.Shared.GameObjects;
+using Robust.Shared.GameObjects.Components.Renderable;
+using Robust.Shared.Interfaces.GameObjects;
+using Robust.Shared.Interfaces.Network;
+using Robust.Shared.IoC;
+using Robust.Shared.Log;
+
+namespace Content.Client.GameObjects.Components.Nutrition
+{
+    [RegisterComponent]
+    public class HungerUI : SharedHungerComponent
+    {
+        private StatusEffectsUI _ui;
+
+        private TextureRect _hungerStatusRect;
+
+#pragma warning disable 649
+        // Required dependencies
+        [Dependency] private readonly IResourceCache _resourceCache;
+#pragma warning restore 649
+
+        public override void OnRemove()
+        {
+            base.OnRemove();
+
+            _ui?.VBox?.RemoveChild(_hungerStatusRect);
+        }
+
+        public override void Initialize()
+        {
+            base.Initialize();
+            if (Owner.TryGetComponent(out SpeciesUI speciesUi)) {
+                _ui = speciesUi.UI;
```

Comment:
I really don't like the way this is hacked in here. Ideally this should be done via a status effects system.